usr/sbin/registercloudguest

```diff
@@ -90,6 +83,43 @@ def get_products():
 
     return products
 
+# ----------------------------------------------------------------------------
+def register_modules(extensions, products, registered=[]):
+    """Register modules obeying dependencies"""
+    for extension in extensions:
+        arch = extension.get('arch')
+        identifier = extension.get('identifier')
+        version = extension.get('version')
+        triplet = '%s/%s/%s' % (identifier, version, arch)
+        if triplet in products and triplet not in registered:
+            registered.append(triplet)
+            cmd = [
+                register_cmd,
+                '--url',
+                'https://%s' % registration_target.get_FQDN(),
+                '--product',
+                triplet
+            ]
+            if os.path.exists(instance_data_filepath):
+                cmd.append('--instance-data')
+                cmd.append(instance_data_filepath)
+
+            logging.info('Registration: %s' % ' '.join(cmd))
+            p = subprocess.Popen(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE
+            )
+            res = p.communicate()
```

Comment:
Consider more verbose naming, e.g. `resp` or `response` and `proc` instead of `p`.